[5.2] Multilang: Remove restrictions for home language menu items

Hackwar opened the pull request:
### Summary of Changes
The multilanguage system currently requires each home menu item to be in a separate menu, making the setup unnecessarily complex. There is no reason for forcing this behavior and this PR removes that. With this PR, the restriction is removed and you can put all home menu items into one menu. In the backend menu, the menu listing is changed to display the home icon for menus with more than one home menu item.


### Testing Instructions
Setup a multi language website and try to save all home language menu items in one menu.


### Actual result BEFORE applying this Pull Request
Error when saving.


### Expected result AFTER applying this Pull Request
No errors.


### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

Hackwar committed:
Multilang: Remove restrictions for home language menu items

brianteeman commented:
I created a multilingual site with 4 languages and the multilingual sample data. 
I moved all the menu items to one menu and deleted the other menus

That all worked as expected. 

It just left the check in the status module to be updated as the information is no longer correct

![image](https://github.com/user-attachments/assets/96bbf97d-a63d-4e11-bb3c-a33fc00e1866)

coolcat-creations commented:
Tested Successfully but 2 things to consider:
1) Should there be another Sample Data Mode for the menus?
2) When I filter the menu items by language the nesting is not displayed correctly, should there be another filter ?

brianteeman commented:
> When I filter the menu items by language the nesting is not displayed correctly, should there be another filter ?

Can you post a screenshot as it looks ok to me
![image](https://github.com/user-attachments/assets/bfcc3062-9fae-442d-8c8d-97598b40bdeb)

coolcat-creations commented:
It looks like this after filtering altough its nested:
<img width="405" alt="grafik" src="https://github.com/user-attachments/assets/3dd059d0-d8fa-408e-a46d-893052c346ab">

brianteeman commented:
Unable to replicate (or i dont understand)

![image](https://github.com/user-attachments/assets/bdf4928e-17b6-43e1-a982-b867798b1697)

coolcat-creations commented:
indeed - yours is still nested -mine not

brianteeman commented:
> indeed - yours is still nested -mine not

please post the full screenshot

coolcat-creations commented:
Sorry was my mistake, the nested Item was set to "all" 
I am still confused that all is not really "All" but just "All" - I mean if something is set to all it should apply for every other language too.

brianteeman committed:
Update default.php

coolcat-creations commented:
I have tested this item :white_check_mark: successfully on bc7029b57695c26c98cfa0e147dedb3c09454db0<br /><br />Works as expected. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43862">issues.joomla.org/tracker/joomla-cms/43862</a>.</sub>

brianteeman committed:
Update com_languages.ini

brianteeman commented:
I have made a pr to your branch for the multilangstatus issue mentioned above

Hackwar committed:
Merge pull request #64 from brianteeman/patch-5

Remove test for home

brianteeman commented:
I have tested this item :white_check_mark: successfully on 1fd1d4efb7b4550557d0e2d34308e55fb6352b19<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43862">issues.joomla.org/tracker/joomla-cms/43862</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43862">issues.joomla.org/tracker/joomla-cms/43862</a>.</sub>

QuyTon committed:
Merge branch '5.2-dev' into 5.2-langhomemenuitem

pe7er committed:
Merge branch '5.2-dev' into 5.2-langhomemenuitem

muhme committed:
Fix PostgreSQL failure

With the PR there are two exceptions if as database is PostgreSQL is used.
`ERROR: column "m.language" must appear in the GROUP BY clause or be used in an aggregate function`

If the columns in GROUP BY are simple added again, there are multiple
menu entries if multiple home set in one menu.

Solution is to use the aggregat function MAX.

Tested with
* PostgreSQL PDO
* MySQL with MySQLi
* MySQL with MySQL PDO
* MariaDB with MySQL PDO
* MariaDB with MySQLi

On base of
* PR 43862 with Patch Tester
* npm update joomla-cypress 1.1.1
* this PR fix-postgres
* npx cypress run --spec tests/System/integration/install/multi-lang-menu.cy.js
Test Support PR 43862 with
✓ install Japanese language pack  (9784ms)
✓ install Ukrainian language pack (8487ms)
✓ install German language pack (8309ms)
✓ enable plugin 'System - Language filter' (1903ms)
✓ create '2nd menu' (1484ms)
✓ create 1st menu entry de as HOME in '2nd menu' (5601ms)
✓ create 2nd menu entry ja as HOME in '2nd menu' (4936ms)
✓ create 3rd menu entry uk as HOME in '2nd menu' (5392ms)
✓ create '3rd menu' (1434ms)

* checked five times that the HOME symbol is displayed if more than one Home menu entry is set,
  otherwise the correct language (uk-UA, ja-JP or de-DE) is displayed for each of the three entries
  if there is only one menu entry with Home

Hackwar committed:
Merge pull request #65 from muhme/fix-postgres

Fix PostgreSQL failure

Hackwar commented:
We had an issue with PostgreSQL in the admin menu, but thanks to @muhme this is now fixed. @brianteeman @coolcat-creations would you be able to test this again? Basically it would be enough to make sure that the backend menu still functions as expected.

Hackwar commented:
Back to pending.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43862">issues.joomla.org/tracker/joomla-cms/43862</a>.</sub>

brianteeman commented:
I have tested this item :white_check_mark: successfully on 7d82dcd0f9b6de337e4af9948e89e22eafb344f6<br /><br />still ok<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43862">issues.joomla.org/tracker/joomla-cms/43862</a>.</sub>

QuyTon committed:
Merge branch '5.2-dev' into 5.2-langhomemenuitem

fgsw commented:
On a nightly build:

1. Installation Multilingual Sample Data (en-GB, de-DE)
2. Installation Testing Sample Data
3. Move all Menu items of both languages to "Main Menu"

See Icon "Home Default" (white House) in "Main Menu Testing" and "Main Menu":

![Untitled 2](https://github.com/user-attachments/assets/67309fd3-bdb0-45ee-bb53-4bd16967ace7)

Test System Information:
**PHP Built On** 	Darwin Air.local 23.6.0 Darwin Kernel Version 23.6.0: Mon Jul 29 21:13:00 PDT 2024; root:xnu-10063.141.2~1/RELEASE_X86_64 x86_64
**Database Type** 	mysql
**Database Version** 	8.0.35
**Database Collation** 	utf8mb4_unicode_ci
**Database Connection Collation** 	utf8mb4_0900_ai_ci
**Database Connection Encryption** 	**None**
**Database Server Supports Connection Encryption** 	**Yes**
**PHP Version** 	8.3.8
**Web Server** 	Apache/2.4.58 (Unix) OpenSSL/1.1.1u mod_fastcgi/mod_fastcgi-SNAP-0910052141
**WebServer to PHP Interface** 	cgi-fcgi
**Joomla! Version** 	Joomla! 5.2.0-alpha4-dev Development [ Uthabiti ] 23-July-2024 16:01 GMT
**Joomla Backward Compatibility Plugin** 	**Disabled**
**User Agent** 	Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:129.0) Gecko/20100101 Firefox/129.0

Hackwar commented:
@fgsw That is the expected behavior. If a menu contains more than one home menu item, it should show the house icon next to it. We can't really display the different flags, since that would already get ugly with 2 flags.

If you could mark this as tested successfully, then I would try to still get this into beta 1.

fgsw commented:
I have tested this item :white_check_mark: successfully on ea5b52c95b4ee12fdeef646653bb1a839f690b4e<br /><br />@Hackwar Thanks for comment. I thought the icon is only for home-menu-item-all languages.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43862">issues.joomla.org/tracker/joomla-cms/43862</a>.</sub>

Hackwar commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43862">issues.joomla.org/tracker/joomla-cms/43862</a>.</sub>

Hackwar committed:
Merge branch '5.2-dev' into 5.2-langhomemenuitem

bembelimen commented:
Thx

alexandreelise commented:
Awesome work Super Joomlers! Thanks for this feature

Quy commented on libraries/src/Table/Menu.php:
```diff
@@ -258,22 +258,6 @@ public function store($updateNulls = true)
-                )
-                && ($table->language != $this->language)
-            ) {
-                $this->setError(Text::_('JLIB_DATABASE_ERROR_MENU_HOME_NOT_UNIQUE_IN_MENU'));
```
Deprecate string?
